snap/component.go

```diff
@@ -143,6 +146,41 @@ func (c *componentPlaceInfo) MountDescription() string {
 	return fmt.Sprintf("Mount unit for %s, revision %s", c.ContainerName(), c.compRevision)
 }
 
+// ComponentLinkPath returns the path for the symlink for a component for a
+// given snap revision.
+func ComponentLinkPath(cpi ContainerPlaceInfo, snapRev Revision) string {
+	instanceName, compName, _ := strings.Cut(cpi.ContainerName(), "+")
+	compBase := ComponentsBaseDir(instanceName)
+	return filepath.Join(compBase, snapRev.String(), compName)
+}
+
+// ComponentInstallDate returns the "install date" of the component by checking
+// when its symlink was created. We cannot use the mount directory as lstat
+// returns the date of the root of the container instead of the date when the
+// mount directory was created.
+func ComponentInstallDate(cpi ContainerPlaceInfo, snapRev Revision) *time.Time {
```

Comment:
Should this follow the same pattern as `snap.InstallDate`? It returns a `time.Time{}` when we can't figure out the time, rather than a nil pointer.

Reply:
Although I see that you're using a pointer inside of `client.Component`, so it works well with `omitempty`. So this is probably file.